packages/block-editor/src/store/selectors.js

```diff
@@ -2813,12 +2813,13 @@ export function __unstableHasActiveBlockOverlayActive( state, clientId ) {
 
 	const editorMode = __unstableGetEditorMode( state );
 
-	// In zoom-out mode, the block overlay is always active for top level blocks.
-	if (
-		editorMode === 'zoom-out' &&
-		clientId &&
-		! getBlockRootClientId( state, clientId )
-	) {
+	// In zoom-out mode, the block overlay is always active for section level blocks.
+	if ( editorMode === 'zoom-out' ) {
+		const { sectionRootClientId } = unlock( getSettings( state ) );
+		const sectionClientIds = getBlockOrder( state, sectionRootClientId );
+		if ( sectionClientIds?.includes( clientId ) ) {
+			return true;
+		}
 		return true;
```

Comment:
If I'm reading the previous code properly, this should only return true here if clientId is a root block (no parent)

Reply:
Well it's applied if a section container is provided so it should apply if no section container is provided (root level), it's just the fallback no?

Reply:
In b96f5a659f0a6be6fe711b4a0c2d953f2a8d6254 I corrected this as it was always returning true in zoom out for any client id, which is wrong, and I added that in case there is no section root to return true for the top most parent. But that is only here as a safe guard since in `getSectionRootBlock` we always return something, so there always is a root - even if sometimes it's the root element (`''`).